src/main/java/com/matuageorge/technicalmaintenanceschedule/service/TaskServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.matuageorge.technicalmaintenanceschedule.service;
+
+import com.matuageorge.technicalmaintenanceschedule.dto.TaskDto;
+import com.matuageorge.technicalmaintenanceschedule.exception.NotFoundException;
+import com.matuageorge.technicalmaintenanceschedule.exception.ResourceAlreadyExistsException;
+import com.matuageorge.technicalmaintenanceschedule.exception.ValidationException;
+import com.matuageorge.technicalmaintenanceschedule.model.Task;
+import com.matuageorge.technicalmaintenanceschedule.repository.TaskRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class TaskServiceImpl implements TaskService {
+
+    private final TaskRepository taskRepository;
+    private final ModelMapper modelMapper;
+
+    @Autowired
+    public TaskServiceImpl(TaskRepository taskRepository, ModelMapper modelMapper) {
```

Comment:
You can use @AllArgsConstructor annotation to avoid creation of constructor by hand.